openapi/social/social.yaml

```diff
@@ -275,21 +361,57 @@ components:
             messageText:
               type: string
               description: 'The text of the message you’re making. This text is subject to limitations depending on what service you are posting to, such as Tweet length for Twitter. '
-            mediaUrls:
-              description: 'Links to pieces of media you want on the message you’re sending. These can be images, videos, and gifs. Media are subject to limitations depending on the social network you are posting to.'
+            status:
+              type: string
+              enum:
+                - scheduled
+                - posted
+              description: 'A string representing the tag assigned to a post.'
+              example: 'SCHEDULED'
+            media:
               type: array
+              description: 'An array containing the media attached to the post.'
               items:
-                type: string
-            sendAt:
+                type: object
+                properties:
+                  type:
+                    type: string
+                    enum:
+                      - image
+                      - video
+                      - gif
+                    example: 'image'
+                  url:
+                    type: string
+                    example: 'https://www.example.com/image.jpg'
```

Comment:
Does knowing the content type of the URL matter to the people that will be calling the API? 
For the schedule message operation we allowed them to simply send us a list of URLs and we determined the content type while validating them. If we make this change we will have to update the `Create` handler as well.

Reply:
We made this change after `Instagram` enabled posting with multiple type of media files aka `Carousel`. Right now, we do validate it for different networks but we have internally started saving the `mediaType` field with `mediaURL`. 
We will update the `Create` handler to accommodate this change.

Reply:
Also, the current `messages` API is open to trusted testers, but we have not got any to use it. So I am assuming it would be safe to modify the endpoint with new updates.

Reply:
Yes it is safe to change the create API. Would the type be required when creating a message or do we continue to infer it? I think we should infer it and mark it a readonly field.

Reply:
yea, we should be able to infer it. We can mark it as readonly.